core/kernel/generic_pm.c

```diff
@@ -0,0 +1,147 @@
+// SPDX-License-Identifier: BSD-2-Clause
+/*
+ * Copyright (c) 2018, Linaro Limited
+ */
+
+#include <keep.h>
+#include <kernel/generic_pm.h>
+#include <kernel/panic.h>
+#include <mm/core_memprot.h>
+#include <stdbool.h>
+#include <stddef.h>
+#include <stdint.h>
+#include <string.h>
+#include <types_ext.h>
+
+/*
+ * Generic power management support for platforms and drivers.
+ * Core components can register callback for suspend and resume
+ * sequences. Callback can be related to a specific subsystem,
+ * based on an opaque identifier. Callback has some ordering
+ * directives as specified by the PM_CB_LEVEL_xxx value.
+ *
+ * Platform specific power sequence can call pm_suspend_resume()
+ * for a suspend or resume operation and for a target subsystem.
+ * Registered callbacks for that subsystem are called with the
+ * ordering directive.
+ */
+
+static bool __maybe_unused cb_is_valid(int (*cb)(enum pm_op op, void *hdl),
+				       void *hdl)
+{
+	paddr_t cb_phy = virt_to_phys(cb);
+	paddr_t hdl_phy = virt_to_phys(hdl);
+	bool __maybe_unused valid;
+
+	valid = (phys_to_virt(cb_phy, MEM_AREA_TEE_RAM_RX) == cb) &&
```

Comment:
We should have a global `is_unpaged()` helper function instead, and that function should compare the virtual address with suitable symbols from the links script instead.

Reply:
ok, i will look at that.
Regarding the tested value, I intentionally considered secure RAMs area as invalid candidate. Is that OK to you?

Reply:
I guess so. Propose something and we'll take it from there.